crates/dataflow-jit/src/codegen/call.rs

```diff
@@ -1173,6 +1175,103 @@ impl CodegenCtx<'_> {
+            }
+            ColumnType::I16 => {
+                self.parse_scalar_from_string("parse_i16_from_str", expr_id, call, builder);
+            }
+            ColumnType::U32 => {
+                self.parse_scalar_from_string("parse_u32_from_str", expr_id, call, builder);
+            }
+            ColumnType::I32 => {
+                self.parse_scalar_from_string("parse_i32_from_str", expr_id, call, builder);
+            }
+            ColumnType::U64 => {
+                self.parse_scalar_from_string("parse_u64_from_str", expr_id, call, builder);
+            }
+            ColumnType::I64 => {
+                self.parse_scalar_from_string("parse_i64_from_str", expr_id, call, builder);
+            }
+
+            ColumnType::Usize => {
+                if self.usize_is_u64() {
+                    self.parse_scalar_from_string("parse_u64_from_str", expr_id, call, builder);
+                } else {
+                    self.parse_scalar_from_string("parse_u32_from_str", expr_id, call, builder);
+                }
+            }
+            ColumnType::Isize => {
+                if self.isize_is_i64() {
+                    self.parse_scalar_from_string("parse_i64_from_str", expr_id, call, builder);
+                } else {
+                    self.parse_scalar_from_string("parse_i32_from_str", expr_id, call, builder);
+                }
+            }
+
+            ColumnType::F32 => {
+                self.parse_scalar_from_string("parse_f32_from_str", expr_id, call, builder);
+            }
+            ColumnType::F64 => {
+                self.parse_scalar_from_string("parse_f64_from_str", expr_id, call, builder);
+            }
+
+            ColumnType::Date
+            | ColumnType::Timestamp
+            | ColumnType::String
+            | ColumnType::Unit
+            | ColumnType::Ptr => todo!(),
+        }
+    }
+
+    fn parse_scalar_from_string(
+        &mut self,
+        intrinsic: &str,
+        expr_id: ExprId,
+        call: &Call,
+        builder: &mut FunctionBuilder<'_>,
+    ) {
+        let string = self.value(call.args()[0]);
+        let length = self.string_length(string, self.is_readonly(call.args()[0]), builder);
+        let ptr = self.string_ptr(string, builder);
+
+        let ret_ty = call.ret_ty().native_type().unwrap();
+
+        let slot = builder.create_sized_stack_slot(StackSlotData::new(
+            StackSlotKind::ExplicitSlot,
+            ret_ty.size(&self.frontend_config()),
+        ));
+        let slot_ptr = builder.ins().stack_addr(self.pointer_type(), slot, 0);
+
+        let intrinsic = self.imports.get(intrinsic, self.module, builder.func);
+        let errored = builder.call_fn(intrinsic, &[ptr, length, slot_ptr]);
+
+        // FIXME: We should return a two-valued struct to allow users to handle
+        // failed parsing, which is super expected
+        builder.ins().trapnz(errored, TRAP_FAILED_PARSE);
```

Comment:
I think that in general failed parsing will return NULLs in SQL.

Reply:
I figured, I just don't currently have a way to express that well within the ir, I need aggregate return types of some kind so that I can return a `{ T, bool }` or `{ ?T }`